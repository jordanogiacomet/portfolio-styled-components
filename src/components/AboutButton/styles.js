import styled, { css } from 'styled-components';

export const Container = styled.button`
  background: transparent;
  position: relative;
  border: none;
  color: #ADB7BE;
  margin-right: 12px;
  cursor: pointer;
  padding: 0.5rem 1rem;
  &.light{
    color: black;
    ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
  }
  &:hover {
    color: #1E90FF;
  }
  &::after {
    content: '';
    display: block;
    height: 2px;
    background: linear-gradient(to right, #1E90FF, #4169E1, #00008B);
    transition: width 0.3s ease-in-out;
    width: 0;
    position: absolute;
    left: 0;
    bottom: 0;
  }
  &:hover::after {
    width: 100%;
  }
  @media (min-width: 768px) {
    font-size: 2rem;
  }
  @media (max-width: 640px) {
    font-size: 1.6rem;
  }
  ${({ active }) =>
    active &&
    css`
      color: #1E90FF;
      &::after{
        width: 100%;
      }
    `}
`;
