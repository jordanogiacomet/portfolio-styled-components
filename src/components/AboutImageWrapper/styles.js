import styled from 'styled-components';

export const Container = styled.div`
  margin-top: 4px;
  order: 2;
  > img {
  display: block;
  max-width: 100%;
  height: auto;
  border-radius: 10px; /* Ajuste o raio da borda conforme necess√°rio */
  box-shadow: 0 4px 8px rgba(0, 0, 0, 0.1), 0 6px 20px rgba(0, 0, 0, 0.1);
  }
  
  @media (min-width: 768px) {
  margin-top: 0;
  }
`;
