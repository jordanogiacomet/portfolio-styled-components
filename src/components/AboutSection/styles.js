import styled from 'styled-components';

export const Container = styled.section`
  color: white;
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: center;
    margin-top: 80px;
  }
  @media (min-width: 1280px) {
    gap: 50px;

  };
`;
