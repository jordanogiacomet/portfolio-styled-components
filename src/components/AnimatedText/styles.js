import styled from 'styled-components';
import { TypeAnimation } from 'react-type-animation';

export const Container = styled(TypeAnimation)`
  display: inline-block;
  
  &.light{
    color: ${({ theme }) => theme.colors.primaryColor}
  }
  &.dark{
    ${({ theme }) => theme.colors.secondaryColor}
  }
`;