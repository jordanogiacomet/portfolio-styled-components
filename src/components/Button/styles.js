import styled, { css } from 'styled-components';

const getButtonStyles = ({ theme, type }) => {
  const buttonType = theme.button[type] || theme.button.default;
  return css`
    padding: ${buttonType.padding};
    border-radius: ${buttonType.borderRadius};
    color: ${buttonType.textColor};
    background: ${buttonType.background};
    font-weight: bold;
    cursor: pointer; 
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: 150px; /* Ajuste a largura mínima conforme necessário */
    transition: scale ease-in-out 0.2s;
    &:hover {
      transition: scale ease-in-out 0.2s;
      scale: calc(1.03);
    };

    @media (${theme.media.lteMedium}) {
      width: 100%; /* Largura 100% em telas menores */
    }
  `;
};

export const Container = styled.button`
  ${getButtonStyles};
`; 