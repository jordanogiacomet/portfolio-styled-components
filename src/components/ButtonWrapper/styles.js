import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  gap: 1rem;
  flex-wrap: wrap;
  margin-top: 2rem;

  @media (max-width: 768px) {
    flex-direction: column;
  }
`;