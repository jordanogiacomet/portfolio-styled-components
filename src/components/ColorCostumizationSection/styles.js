// src/components/ColorCustomizationSection/styles.js
import styled from 'styled-components';

export const Container = styled.section`
  display: flex;
  flex-direction: column;
  gap: 50px;
  margin-bottom: 50px;
`;
