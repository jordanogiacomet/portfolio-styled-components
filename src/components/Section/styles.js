import styled, { css } from 'styled-components';

const sectionStyles = {
  'hero-section': css`
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    @media(max-width: 400px) {
       > div > h1 {
        font-size: ${({ theme }) => theme.font.sizes.xl} !important;
       }
    }

    @media ${({ theme }) => theme.media.sm} {
      grid-template-columns: repeat(12, 1fr);
      align-items: center;
      gap: 5rem;
    }
  `,
  'about-section': css`
  color: ${({ theme }) => theme.colors.white};
  padding: 50px 0px;
  display: flex;
  flex-direction: column;
  @media ${({theme}) => theme.media.md} {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 50px;
    align-items: start; /* Altere para start para evitar centralização vertical */
    margin-top: 80px;
  }
  @media ${({theme}) => theme.media.xl} {
    gap: 50px;
  };
`,
  'projects-section': css`
    margin-top: 30px;
    margin-bottom: 240px;
    text-align: left;
    display: flex;
    flex-direction: column;
    height: auto;
    @media ${({theme}) => theme.media.md}  {
      margin-top: 0;
    }
    @media ${({theme}) => theme.media.lg}  {
       margin-bottom: 100px;
    };
  `,
  'email-section': css`
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    margin-bottom: 100px;
    @media(max-width: 640px) {
      margin-bottom: 30px;
    }

    @media ${({ theme }) => theme.media.lg} {
      grid-template-columns: repeat(12, 1fr);
      align-items: center;
      justify-content: space-between;
      gap: 5rem;
    }
  `,
};

export const Container = styled.section`
  ${({ type }) => sectionStyles[type] || ''};
`;
