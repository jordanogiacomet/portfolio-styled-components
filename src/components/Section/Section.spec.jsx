import { renderTheme } from '../../styles/render-theme';
import { Section } from '.';
import { screen } from '@testing-library/react';
import 'jest-styled-components';

describe('<Section />', () => {
  it('should render Section with children', () => {
    renderTheme(
      <Section type="hero-section">
        <div>Hero Section Content</div>
      </Section>
    );
    const element = screen.getByText('Hero Section Content');

    expect(element).toBeInTheDocument();
  });

  it('should apply the correct styles for hero-section', () => {
    const { container } = renderTheme(
      <Section type="hero-section">
        <div>Hero Section Content</div>
      </Section>
    );

    const wrapper = container.firstChild;
    expect(wrapper).toHaveStyleRule('display', 'grid');
    expect(wrapper).toHaveStyleRule('grid-template-columns', 'repeat(1, 1fr)');
  });

  it('should apply the correct styles for about-section', () => {
    const { container } = renderTheme(
      <Section type="about-section">
        <div>About Section Content</div>
      </Section>
    );

    const wrapper = container.firstChild;
    expect(wrapper).toHaveStyleRule('color', 'white');
    expect(wrapper).toHaveStyleRule('padding', '50px 0px');
    expect(wrapper).toHaveStyleRule('display', 'flex');
    expect(wrapper).toHaveStyleRule('flex-direction', 'column');
  });

  it('should apply the correct styles for projects-section', () => {
    const { container } = renderTheme(
      <Section type="projects-section">
        <div>Projects Section Content</div>
      </Section>
    );

    const wrapper = container.firstChild;
    expect(wrapper).toHaveStyleRule('margin-top', '30px');
    expect(wrapper).toHaveStyleRule('margin-bottom', '240px');
    expect(wrapper).toHaveStyleRule('text-align', 'left');
    expect(wrapper).toHaveStyleRule('display', 'flex');
    expect(wrapper).toHaveStyleRule('flex-direction', 'column');
  });

  it('should apply the correct styles for email-section', () => {
    const { container } = renderTheme(
      <Section type="email-section">
        <div>Email Section Content</div>
      </Section>
    );

    const wrapper = container.firstChild;
    expect(wrapper).toHaveStyleRule('display', 'grid');
    expect(wrapper).toHaveStyleRule('grid-template-columns', 'repeat(1, 1fr)');
    expect(wrapper).toHaveStyleRule('margin-bottom', '100px');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <Section type="hero-section">
        <div>Hero Section Content</div>
      </Section>
    );

    expect(container).toMatchSnapshot();
  });
});
