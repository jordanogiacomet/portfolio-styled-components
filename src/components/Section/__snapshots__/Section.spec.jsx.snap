// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should match snapshot 1`] = `
.c0 {
  display: grid;
  grid-template-columns: repeat(1, 1fr);
}

@media (min-width: 640px) {
  .c0 {
    grid-template-columns: repeat(12, 1fr);
    align-items: center;
    gap: 5rem;
  }
}

<div>
  <section
    class="c0"
    type="hero-section"
  >
    <div>
      Hero Section Content
    </div>
  </section>
</div>
`;
